### ISSUE TITLE
bug: Tab Icon Jumps (flicker) when pressing on tabs in footer

### ISSUE THREAD

**Issue by lordgreg** [driftyco/ionic#4563]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>ios</span> <span ionic-platform-version>8</span>  <span ionic-webview>webview</span>

<span ionic-description>I don't know if someone already posted a bug here or not. Opening a simple tab-bar example form Ionic on iOS9 brings a bug when pressing tabs in footer. They move when an icon transition is made. 

There's already a forum post on official Ionic forums here: http://forum.ionicframework.com/t/tab-icon-jumps/34316

Person `buunguyen` also recorded the issue which can be seen here: http://jmp.sh/0A5ql5d

There are currently two options to fix the issue: 
1. Either edit the `ionic.bundle.js` or
2. add new `.config` to your app which will replace the part of the code where the transition happens. 

Person `buunguyen` also posted a code in ES6 format, here's the same in ES5:

```
angular
.module('ionic')
.config(['$provide', function ($provide) {
  $provide.decorator('ionTabNavDirective', ['$delegate', function ($delegate) {
    var directive = $delegate[0]
    directive.template = '<a ng-class="{\'tab-item-active\': isTabActive(), \'has-badge\':badge, \'tab-hidden\':isHidden()}" ng-disabled="disabled()" class="tab-item"> <span class="badge {{badgeStyle}}" ng-if="badge">{{badge}}</span> <i class="icon {{getIconOn()}}"></i> <span class="tab-title" ng-bind-html="title"></span></a>'
    return $delegate
  }])
}])
```

We have also tried to get the latest #master but that didn't solve anything.

:+1:  to developers of Ionic :)</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @lordgreg!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=4563). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=4563)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by simplesmiler** [comment]
Duplicate of https://github.com/driftyco/ionic/issues/4395
TLDR: Apply the patch from http://blog.ionic.io/ios-9-potential-breaking-change/

**Comment by lordgreg** [comment]
Thank you.

**Comment by lordgreg** [comment]
Actually no @simplesmiler.

We have implemented ionic Patch and the tab jumping is still there.

**Comment by markahon** [comment]
There was a duplicate bug #4006, where I commented about a CSS workaround I used:

```
/* For a moment there's two <i>-elements inside tab-item, don't allow both to be shown. */
.tabs .tab-item i:nth-of-type(2) {
  display: none;
}
```

This doesn't fix the underlyng problem, but prevents the icon flickering.

**Comment by sirJiggles** [comment]
+1 @markahon, I also applied the patch and had to disable the second icon. Merci

**Comment by brienpafford** [comment]
Thanks @markahon, this css workaround is gold.

**Comment by edmondchui** [comment]
@markahon Thanks for the fix. It works well on browser. I need to use

```css
display: none !important;
```
to make it work on iOS 9 as well.

**Comment by jgw96** [comment]
Hello! Im glad you all found a workaround! Thanks @markahon ! Are you all still experiencing this issue?

**Comment by markahon** [comment]
Seems to me that at least on ionic version `1.2.4` the flicker is gone when I remove the css workaround.

**Comment by jgw96** [comment]
Awesome! Im glad to hear that! Ill be closing this issue for now as it looks like it was resolved in 1.2.4. Thanks again!

**Comment by perry** [comment]
This is still an issue in the latest nightly (`v1.2.4-nightly-2826`) on iOS 9.3. Applying the patch above fixes it.

**Comment by M4v3R** [comment]
I can confirm that I have this issue as well on 1.2.4.

**Comment by ladmerc** [comment]
Yup, me too
